### PR TITLE
217 add a new instruction type to combine execute trx from account data iterative and continue

### PR DESCRIPTION
**Pull request by vasiliy-zaznobin**
Added a test to check deploy with the new type instruction 0x0E without the Continue type instruction 0x0A

### CODE REVIEW

File: evm_loader/program/src/entrypoint.rs [hunk 1]
```diff
@@ -565,6 +565,114 @@ fn process_instruction<'a>(
...
+
+                    applies_and_invokes(
+                        program_id,
+                        &mut account_storage,
+                        accounts,
+                        Some(operator_sol_info),
+                        call_results)?;
+
+                    storage.unblock_accounts_and_destroy(program_id, trx_accounts)?;
+                }
```
Comment (otselnik) [inline evm_loader/program/src/entrypoint.rs:671]: Extract function from these lines. Exact same text in `Continue` and in `PartialCallOrContinueFromRawEthereumTX`

File: evm_loader/program/src/entrypoint.rs [hunk 2]
```diff
@@ -565,6 +565,114 @@ fn process_instruction<'a>(
...
+                return Err!(ProgramError::InvalidAccountData);
+            }
+
+            let caller = verify_tx_signature(signature, unsigned_msg).map_err(|e| E!(ProgramError::MissingRequiredSignature; "Error={:?}", e))?;
+            let trx: UnsignedTransaction = rlp::decode(unsigned_msg).map_err(|e| E!(ProgramError::InvalidInstructionData; "DecoderError={:?}", e))?;
+            let trx_gas_limit = u64::try_from(trx.gas_limit).map_err(|e| E!(ProgramError::InvalidInstructionData; "e={:?}", e))?;
+            let trx_gas_price = u64::try_from(trx.gas_price).map_err(|e| E!(ProgramError::InvalidInstructionData; "e={:?}", e))?;
+            // if trx_gas_price < 1_000_000_000_u64 {
+            //     return Err!(ProgramError::InvalidArgument; "trx_gas_price < 1_000_000_000_u64: {} ", trx_gas_price);
+            // }
```
Comment (otselnik) [inline evm_loader/program/src/entrypoint.rs:592]: These lines should only be executed if it is in `begin` state.

File: evm_loader/program/src/entrypoint.rs (hunk 1 above)
Comment (vasiliy-zaznobin) [reply evm_loader/program/src/entrypoint.rs]: Fixed

File: evm_loader/program/src/entrypoint.rs (hunk 2 above)
Comment (vasiliy-zaznobin) [reply evm_loader/program/src/entrypoint.rs]: Fixed

File: evm_loader/program/src/entrypoint.rs [hunk 3]
```diff
@@ -565,6 +565,114 @@ fn process_instruction<'a>(
...
+                    storage_info,
+                    system_info)?;
+
+                if trx.to.is_some() {
+                    do_partial_call(&mut storage, step_count, &account_storage, trx_accounts, trx.call_data, trx.value, trx_gas_limit)?;
+                } else {
+                    do_partial_create(&mut storage, step_count, &account_storage, trx_accounts, trx.call_data, trx.value, trx_gas_limit)?;
+                }
+
+                storage.block_accounts(program_id, trx_accounts)
```
[the remark this answers is not in the document]
Comment (vasiliy-zaznobin) [reply evm_loader/program/src/entrypoint.rs]: Fixed

File: evm_loader/program/src/entrypoint.rs [hunk 4]
```diff
@@ -496,15 +496,49 @@ fn process_instruction<'a>(
...
+            debug_print!("Execute iterative transaction from account data or continue");
+            let holder_info = next_account_info(account_info_iter)?;
+            let storage_info = next_account_info(account_info_iter)?;
+            let operator_sol_info = next_account_info(account_info_iter)?;
+            let collateral_pool_sol_info = next_account_info(account_info_iter)?;
+            let operator_eth_info = next_account_info(account_info_iter)?;
+            let user_eth_info = next_account_info(account_info_iter)?;
+            let system_info = next_account_info(account_info_iter)?;
+            let holder_data = holder_info.data.borrow();
+            let (unsigned_msg, signature) = get_transaction_from_data(&holder_data)?;
```
Comment (otselnik) [inline evm_loader/program/src/entrypoint.rs:518]: Move this line to begin block

**Review by otselnik — APPROVED** [review]

**Pull request merged**